modules/search-service-impl/src/main/java/org/opencastproject/search/impl/SearchServiceIndex.java

```diff
@@ -274,6 +296,31 @@ private void indexMediaPackage(MediaPackage mediaPackage, AccessControlList acl,
       }).filter(Objects::nonNull).collect(Collectors.toList());
     }
 
+    // Add custom roles to the ACL
+    // This allows users with a role of the form ROLE_EPISODE_<ID>_<ACTION> to access the event through the index
+    if (episodeIdRole) {
+      Set<AccessControlEntry> customEntries = new HashSet<>();
+      customEntries.add(new AccessControlEntry(getEpisodeRoleId(mediaPackageId, "READ"), "read", true));
+      customEntries.add(new AccessControlEntry(getEpisodeRoleId(mediaPackageId, "WRITE"), "write", true));
+
+      ResourceListQuery query = new ResourceListQueryImpl();
+      if (listProvidersService.hasProvider("ACL.ACTIONS")) {
+        Map<String, String> actions = new HashMap<>();
+        try {
+          actions = listProvidersService.getList("ACL.ACTIONS", query, true);
+        } catch (ListProviderException e) {
+          throw new SearchException("Listproviders not loaded. " + e);
+        }
+        for (String action : actions.keySet()) {
+          customEntries.add(
+              new AccessControlEntry(getEpisodeRoleId(mediaPackageId, action), action, true));
+        }
+      }
+
+      AccessControlList customRoles = new AccessControlList(new ArrayList<>(customEntries));
+      acl = customRoles.merge(acl);
```

Comment:
This looks very very similar to the code in `ElasticsearchIndex::updateEventAclWithCustomRoles`

Reply:
`ElasticsearchIndex::updateEventAclWithCustomRoles` is gone now, so this is arguably outdated.